src/Data/List.agda

```diff
@@ -11,7 +11,35 @@
 
 module Data.List where
 
+import Data.List.NonEmpty.Base as List⁺
+import Data.List.Properties as List
+open import Function.Base using (_∘_)
+open import Relation.Binary.PropositionalEquality using (refl; cong₂; _≗_)
+
 ------------------------------------------------------------------------
 -- Types and basic operations
 
-open import Data.List.Base public
+open import Data.List.Base public hiding (scanr)
+
+------------------------------------------------------------------------
+-- scanr
+
+module _ {a b} {A : Set a} {B : Set b} (f : A → B → B) (e : B) where
+
+  open List⁺ using (List⁺; _∷_; toList)
+
+-- definition
+
+  scanr : List A → List B
+  scanr = toList ∘ go where
```

Comment:
love how this definition makes it clear that `scanr` is forgetful, i.e. drops information it learned along the way!

Reply:
... indeed ;-)

But to refactor it (once again), so that `go` as `scanr⁺` it (properly?!) belongs in `Data.List.NonEmpty`, and hence the `tails` property could be expressed in `Data.List.NonEmpty.Properties`, then means we *also* need to refactor `tails` (as `tails⁺`, for example)... etc. lots of knock-on ... bureaucracy. 

Suggestions as to how to "find the way out of the bottle"?